File: src/main/java/com/hedvig/backoffice/services/account/AccountService.java
package com.hedvig.backoffice.services.account;

import com.hedvig.backoffice.graphql.types.AccountEntryInput;
import com.hedvig.backoffice.services.account.dto.AccountDTO;
import com.hedvig.backoffice.services.account.dto.ApproveChargeRequestDto;
import com.hedvig.backoffice.services.account.dto.SchedulerStateDto;

import java.util.List;

public interface AccountService {
  AccountDTO getAccount(String memberId);

  List<AccountDTO> batchFindCurrentBalances(List<String> memberIds);

  void addAccountEntry(String memberId, AccountEntryInput accountEntryInput, String addedBy);

  void backfillSubscriptions(String memberId, String backfilledBy);

  List<SchedulerStateDto> subscriptionSchedulesAwaitingApproval(ChargeStatus status);

  void addApprovedSubscriptions(List<ApproveChargeRequestDto> requestBody, String approvedBy);
}

File: src/main/java/com/hedvig/backoffice/services/account/AccountService.kt
package com.hedvig.backoffice.services.account

import com.hedvig.backoffice.graphql.types.account.AccountEntryInput
import com.hedvig.backoffice.graphql.types.account.MonthlyEntryInput
import com.hedvig.backoffice.services.account.dto.*
import java.util.*

interface AccountService {
    fun batchFindCurrentAccounts(memberIds: List<String>): List<AccountDTO?>
    fun addAccountEntry(memberId: String, accountEntryInput: AccountEntryInput, addedBy: String)
    fun addMonthlyEntry(memberId: String, monthlyEntryInput: MonthlyEntryInput, addedBy: String)
    fun backfillSubscriptions(memberId: String, backfilledBy: String)
    fun subscriptionSchedulesAwaitingApproval(status: ChargeStatus): List<SchedulerStateDto>
    fun addApprovedSubscriptions(requestBody: List<ApproveChargeRequestDto>, approvedBy: String)
    fun getNumberFailedCharges(memberId: String): NumberFailedChargesDto
    fun getMonthlyEntriesForMember(memberId: String): List<MonthlyEntryDto>
    fun removeMonthlyEntry(id: UUID, removedBy: String)
}
